Cannot compile AUMIDIEvent due to its struct meta-data.

#### Issue details

The robovm fails to build our framework when parsing `org.robovm.apple.audiotoolbox.AUMIDIEvent`. The error message is

> Duplicate @StructMember(1) in @Packed structs (union in packed structs is not supported

#### Reproduction steps/code

```shell
git clone git@github.com:ecsec/open-ecard.git
cd open-ecard
# The following hash is currently origin/fix_ios_packaging
git checkout d65a0bbe008153d88de3b4818ffd405c5315bd25
mvn clean install -DskipTests -pl :ios-core -am
cd packager/ios-framework
mvn clean install -e
```

#### Configuration

**Build Tools:**
- [ ] IDEA plugin
- [ ] Eclipse plugin
- [ ] Gradle plugin
- [x] Maven plugin
 
**Versions:**

_Please provide the version of RoboVM, XCode and JDK used_
- Robovm: 2.3.8-SNAPSHOT (freshly downloaded 20/09/2019)
- XCode: Xcode-beta 6
- JDK: 11.0.2

**Build Targets:**

- framework

---

#### Stacktrace
```java
[INFO] Compiling org.robovm.apple.audiotoolbox.AUInstrumentType (ios arm64 release)                                                                                                                                                                                  [29/43026]
[INFO] Compiling org.robovm.apple.audiotoolbox.AUMIDIEvent (ios arm64 release)         
[INFO] ------------------------------------------------------------------------        
[INFO] BUILD FAILURE                                                                                       
[INFO] ------------------------------------------------------------------------                            
[INFO] Total time:  08:50 min                                                                              
[INFO] Finished at: 2019-09-20T10:47:56+02:00                                                              
[INFO] ------------------------------------------------------------------------                            
[ERROR] Failed to execute goal com.mobidevelop.robovm:robovm-maven-plugin:2.3.8-SNAPSHOT:install (default) on project ios-framework: Execution default of goal com.mobidevelop.robovm:robovm-maven-plugin:2.3.8-SNAPSHOT:install failed: Duplicate @StructMember(1) in @Packed structs (union in packed structs is not supported -> [Help 1]                                              
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.mobidevelop.robovm:robovm-maven-plugin:2.3.8-SNAPSHOT:install (default) on project ios-framework: Execution default of goal com.mobidevelop.robovm:robovm-maven-plugin:2.3.8-SNAPSHOT:instal
l failed: Duplicate @StructMember(1) in @Packed structs (union in packed structs is not supported           
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)       
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)                                
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)                     
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)                       
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)                             
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)                              
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)                                              
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)                               
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)             
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)   
    at java.lang.reflect.Method.invoke (Method.java:566)                                                 
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)    
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)                                          
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)                                                      
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default of goal com.mobidevelop.robovm:robovm-maven-plugin:2.3.8-SNAPSHOT:install failed: Duplicate @StructMember(1) in @Packed structs (union in packed structs is not supported
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:148)                              
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)                                                
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)                                                
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)                                                
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)                                        
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)                                                                 
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)                                                                 
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)                                                                   
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)                                                                       
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)                                                                        
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)                                                                          
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)                                                           
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)                                         
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)                                 
    at java.lang.reflect.Method.invoke (Method.java:566)                                                                               
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)                                            
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)                                                    
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)                                          
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)                                                      
Caused by: java.lang.IllegalArgumentException: Duplicate @StructMember(1) in @Packed structs (union in packed structs is not supported
    at org.robovm.compiler.BroMethodCompiler.getStructType (BroMethodCompiler.java:690)                                                
    at org.robovm.compiler.BroMethodCompiler.getStructType (BroMethodCompiler.java:568)                                                
    at org.robovm.compiler.StructMemberMethodCompiler.reset (StructMemberMethodCompiler.java:63)                                       
    at org.robovm.compiler.ClassCompiler.compile (ClassCompiler.java:783)                                                              
    at org.robovm.compiler.ClassCompiler.compile (ClassCompiler.java:297)                                                              
    at org.robovm.compiler.AppCompiler.compile (AppCompiler.java:353)                                                                  
    at org.robovm.compiler.AppCompiler.compile (AppCompiler.java:436)                                                                  
    at org.robovm.compiler.AppCompiler.compile (AppCompiler.java:500)                                                                  
    at org.robovm.compiler.AppCompiler.build (AppCompiler.java:962)
    at org.robovm.maven.plugin.AbstractRoboVMBuildMojo.execute (AbstractRoboVMBuildMojo.java:66)                                       
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)                              
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)                                                
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)                                                
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)                                                
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)                                        
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)                                                                 
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)                                                                 
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)                                                                   
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)                                                                       
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)                                                                        
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)                                                                          
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)                                                           
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)                                         
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)                                 
    at java.lang.reflect.Method.invoke (Method.java:566)                                                                               
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)                                            
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)                                                    
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)                                          
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
```

--- a/compiler/rt/robovm/src/main/java/org/robovm/rt/bro/ValuedEnum.java
+++ b/compiler/rt/robovm/src/main/java/org/robovm/rt/bro/ValuedEnum.java
@@ -141,7 +141,8 @@ public interface ValuedEnum {
                     + value + " (0x" + Long.toHexString(value) + ") found in " 
                     + enumType.getName());
         }
-        
+
+        @MarshalsValue
         public static long toNative(ValuedEnum v, long flags) {
             return v.value();
         }
@@ -156,7 +157,8 @@ public interface ValuedEnum {
         public static <T extends Enum<T> & ValuedEnum> ValuedEnum toObject(Class<T> cls, @MachineSizedSInt long value, long flags) {
             return AsLongMarshaler.toObject(cls, value, flags);
         }
-        
+
+        @MarshalsValue
         public static @MachineSizedSInt long toNative(ValuedEnum v, long flags) {
             return v.value();
         }
@@ -171,7 +173,8 @@ public interface ValuedEnum {
         public static <T extends Enum<T> & ValuedEnum> ValuedEnum toObject(Class<T> cls, @MachineSizedUInt long value, long flags) {
             return AsLongMarshaler.toObject(cls, value, flags);
         }
-        
+
+        @MarshalsValue
         public static @MachineSizedUInt long toNative(ValuedEnum v, long flags) {
             return v.value();
         }

